tests/by-util/test_tail.rs

```diff
@@ -1512,7 +1540,7 @@ fn test_retry8() {
 }
 
 #[test]
-#[cfg(all(unix, not(any(target_os = "android", target_vendor = "apple"))))] // FIXME: make this work not just on Linux
+#[cfg(all(not(target_os = "windows"), not(target_os = "freebsd")))] // FIXME: for currently not working platforms
```

Comment:
This test is problematic. It has been observed to sometimes fail on MinRustV and now macOS.
https://github.com/uutils/coreutils/issues/3929
Until this is fixed I would not relax the cfg.

Reply:
won't fix

Reply:
Are you aware that this comes across as quite rude? Why is this a won't fix? @jhscheer brings up a valid problem with your PR, so that needs to be fixed before we can merge this. I'm unresolving this conversation until it is actually resolved.

Reply:
It's a won't fix because this is actually valuable information about the intermittenly failing tests. Additionally, these intermittently failing tests are not part of this pr nor the refactoring on which this pr is based on. You may wish to disable the test until it is fixed, but I would suggest another pr for this.

Reply:
But this PR enables the test on mac, which is not correct. So this PR should not introduce that change in the first place.

Reply:
The retry9 test is failing intermittently, but not because of the refactoring pr. It also does on macos but now you can notice this fact and maybe it helps figuring out the problem. The test usually passes and if not then because of #3929.

Reply:
We try to keep intermittent failure to a minimum, because it affects everyone's CI for PRs. We can't accept PRs that knowingly introduce more intermittent failures.

The refactoring PR has nothing to do with this. There is a test that fails (intermittently) and this PR enables it, which is a problem and that is why this PR has to just not do that.

> It also does on macos but now you can notice this fact and maybe it helps figuring out the problem.

We now indeed know this and that is helpful, but that's not a valid reason that we should ignore the problem here. We're not asking you to fix anything, just to revert the change that you made for this specific test.

Reply:
I'm not sure this is going into the right direction, but if dsabling this test on macos is all you want, it's done.